src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -464,6 +464,15 @@ public boolean assignCitizen(final CitizenData citizen)
                   .setBaseValue(SharedMonsterAttributes.ARMOR.getDefaultValue() + getDefenceBonus());
             }
             colony.getCitizenManager().calculateMaxCitizens();
+
+            // Set new home, since guards are housed at their workerbuilding.
+            final AbstractBuilding building = citizen.getHomeBuilding();
+            if (building != null && !(building instanceof AbstractBuildingGuards))
```

Comment:
We should probably do a check if !building.getId.equals(this.getId)

Reply:
the  !(building instanceof AbstractBuildingGuards) already includes it since "this" is always a guard building

Reply:
I mean, to compare if the building is really THIS building. (not only same type)

Reply:
ah that way, you're right, should do that to move guards from one building to another